wandb/sdk/wandb_settings.py

```diff
@@ -1864,6 +1871,16 @@ def _apply_init(self, init_settings: Dict[str, Union[str, int, None]]) -> None:
                         "Tried to auto resume run with "
                         f"id {resume_run_id} but id {self.run_id} is set.",
                     )
+
+        # handle rewind logic
+        if self.resume_from is not None:
```

Comment:
should it be in it's own function? as these not really the same right?

Reply:
factored it out